pkg/ingester/ingester.go

```diff
@@ -1414,8 +1409,9 @@ func (i *Ingester) LabelNames(ctx context.Context, req *client.LabelNamesRequest
 }
 
 // MetricsForLabelMatchers implements IngesterServer.
-func (i *Ingester) MetricsForLabelMatchers(ctx context.Context, req *client.MetricsForLabelMatchersRequest) (*client.MetricsForLabelMatchersResponse, error) {
-	if err := i.checkRunning(); err != nil {
+func (i *Ingester) MetricsForLabelMatchers(ctx context.Context, req *client.MetricsForLabelMatchersRequest) (resp *client.MetricsForLabelMatchersResponse, mlErr error) {
+	defer func() { mlErr = i.handleReadError(mlErr) }()
```

Comment:
Same comment as before, i.e. I think that the returned error should be named `err` to avoid confusion and potential programming mistakes.